packages/flutter/lib/src/material/theme.dart

```diff
@@ -123,7 +127,8 @@ class Theme extends StatelessWidget {
         return null;
       return inheritedTheme.theme.data;
     }
-    return (inheritedTheme != null) ? inheritedTheme.theme.data : _kFallbackTheme;
+    final ThemeData baseTheme = (inheritedTheme != null) ? inheritedTheme.theme.data : _kFallbackTheme;
+    return ThemeData.localize(context, baseTheme);
```

Comment:
I would probably put the call to `MaterialLocalizations.of(context)` here too, so that all the inherited-widget logic is in one place.